controllers/drplacementcontrol_controller.go

```diff
@@ -808,17 +808,20 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return r.reconcileDRPCInstance(d, logger)
 }
 
-func (r *DRPlacementControlReconciler) setProgressionAndUpdate(
-	ctx context.Context, drpc *rmn.DRPlacementControl, progressionStatus rmn.ProgressionStatus,
+func (r *DRPlacementControlReconciler) setDeletionStatusAndUpdate(
+	ctx context.Context, drpc *rmn.DRPlacementControl,
 ) error {
-	updated := updateDRPCProgression(drpc, progressionStatus, r.Log)
+	updated := updateDRPCProgression(drpc, rmn.ProgressionDeleting, r.Log)
+	drpc.Status.Phase = rmn.Deleting
 
 	if updated {
 		if err := r.Status().Update(ctx, drpc); err != nil {
 			return fmt.Errorf("failed to update DRPC status: (%w)", err)
 		}
 	}
 
+	updateDRPCProgression(drpc, rmn.ProgressionDeleted, r.Log)
```

Comment:
On a subsequent reconcile if there is still and error (or in progress) from `processDeletion` we would find `updated` as false and that would cause us to set the progression here to `Deleted` which would be incorrect. This should be removed from here.

Ideally, just [before](https://github.com/RamenDR/ramen/blob/39485b5adc51671842534660e287ce07503542fd/controllers/drplacementcontrol_controller.go#L1179) we remove the finalizer we should update the Progression to "Deleted"
- This helps in case the DRPC lingers around a bit after our finalizer removal for a user to note that we have progressed and completed
- Has to be set before removing the finalizer (or rather with the same Update call), as otherwise the API server may garbage collect the resource before we get to update it

Reply:
moved before removing the finalizer